spec/factories/cart_products.rb

```diff
@@ -0,0 +1,13 @@
+FactoryBot.define do
+  factory :cart_product do
+    cart
+    product
+    quantity { 0 }
+    total_amount { 0 }
+  end
+
+  trait :with_cart_and_product do
+    cart { create(:cart, :with_user) }
+    product { create(:product) }
+  end
```

Comment:
It is a question that I was asking myself in the past since if we need to build the object, I wanted to leave everything as clean as possible. I have updated it to avoid creating a trait.